src/main/java/fr/ospiea/oscra/restsecurity/RESTAuthenticationEntryPoint.java

```diff
@@ -0,0 +1,22 @@
+package fr.ospiea.oscra.restsecurity;
+
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.AuthenticationEntryPoint;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * Created by taozheng on 27/09/2016.
+ */
+@Component
+public class RESTAuthenticationEntryPoint implements AuthenticationEntryPoint {
+    @Override
+    public void commence(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, AuthenticationException e) throws IOException, ServletException {
```

Comment:
Exactly. In fact, I just customize the form login.
Because the default form login, when a user does not have the permission to visit the service, the default form login will return a status code 301 in order to redirect to the login page. But in our case, the redirection should be done in the front-end, the back-end should return a status code 401 (unauthorized) instead of code status code 301 (redirection).